src/AppInstallerRepositoryCore/Microsoft/PortableIndex.cpp

```diff
@@ -1,13 +1,20 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 #include "pch.h"
-#include "PortableIndex.h"
+#include "Public/winget/PortableIndex.h"
+#include "Microsoft/Schema/IPortableIndex.h"
+#include "Microsoft/Schema/Portable_1_0/PortableTable.h"
 #include <winget/SQLiteStorageBase.h>
 #include "Schema/Portable_1_0/PortableIndexInterface.h"
-#include "winget/Filesystem.h"
+#include <winget/Filesystem.h>
 
 namespace AppInstaller::Repository::Microsoft
 {
+    PortableIndex::PortableIndex(PortableIndex&&) = default;
+    PortableIndex& PortableIndex::operator=(PortableIndex&&) = default;
+
+    PortableIndex::~PortableIndex() = default;

```

Comment:
Just curious, is there particular reason we move these default constructor/destructor here where in other places we just define "= default" in header? Or this is the preferred convention in future?

Reply:
It is part of [pImpl](https://en.cppreference.com/w/cpp/language/pimpl); the destruction of the smart pointer has to happen in the compilation unit that has access to the destructor.  Leaving it in the header requires that the caller can do that, but moving it to the .cpp moves that requirement to the implementation.  This allows the `IPortableIndex` type to simply be a forward declaration for the public header and only a complete type in the .cpp.

This is not new convention, but application of a specific technique for this purpose.